test/e2e-test/suite_test.go

```diff
@@ -100,6 +107,30 @@ var _ = BeforeSuite(func(done Done) {
 	// For some reason, traitDefinition is created as a Cluster scope object
 	Expect(k8sClient.Create(context.Background(), &manualscalertrait)).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
 	By("Created manual scalar trait definition")
+
+	// Create manual scaler trait definition with spec.extension field
+	definitionExtension := DefinitionExtension{
+		Alias: "ManualScaler",
+	}
+	in := new(runtime.RawExtension)
+	in.Raw, _ = json.Marshal(definitionExtension)
+
+	extendedmanualscalertrait = v1alpha2.TraitDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "manualscalertraits-extended.core.oam.dev",
+			Labels: map[string]string{"trait": "manualscalertrait"},
+		},
+		Spec: v1alpha2.TraitDefinitionSpec{
+			WorkloadRefPath: "spec.workloadRef",
+			Reference: v1alpha2.DefinitionReference{
+				Name: "manualscalertraits-extended.core.oam.dev",
+			},
+			Extension: in,
```

Comment:
The extension is optional, why do we need to add it in the e2e test?

Reply:
We are checking whether it's worked or not